listener/src/event_handler/auction_settlement_handler.rs

```diff
@@ -0,0 +1,211 @@
+use super::*;
+
+use failure::Error;
+use slog::Logger;
+use std::sync::Arc;
+
+use dfusion_core::database::DbInterface;
+use dfusion_core::models::Deserializable;
+use dfusion_core::models::util::{PopFromLogData, ToValue};
+use dfusion_core::models::Solution;
+
+use graph::components::ethereum::EthereumBlock;
+use graph::components::store::EntityOperation;
+use graph::data::store::Entity;
+
+use web3::types::{H256, U256};
+use web3::types::{Log, Transaction};
+
+use std::fmt;
+
+use super::EventHandler;
+use super::util;
+
+
+#[derive(Clone)]
+pub struct AuctionSettlementHandler {
+    store: Arc<DbInterface>,
+}
+
+impl AuctionSettlementHandler {
+    pub fn new(store: Arc<DbInterface>) -> Self {
+        AuctionSettlementHandler {
+            store
+        }
+    }
+}
+
+impl Debug for AuctionSettlementHandler {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "AuctionSettlementHandler")
+    }
+}
+
+
+impl EventHandler for AuctionSettlementHandler {
+    fn process_event(
+        &self,
+        logger: Logger,
+        _block: Arc<EthereumBlock>,
+        _transaction: Arc<Transaction>,
+        log: Arc<Log>,
+    ) -> Result<Vec<EntityOperation>, Error> {
+        let mut event_data: Vec<u8> = log.data.0.clone();
+        info!(logger, "Parsing Auction Settlement event from {} bytes: {:?}", event_data.len(), event_data);
+        let auction_id = U256::pop_from_log_data(&mut event_data);
+        let state_index = U256::pop_from_log_data(&mut event_data).saturating_sub(U256::one());
+        let new_state_hash = H256::pop_from_log_data(&mut event_data);
+
+        // Strange info coming with packed bytes
```

Comment:
Can we make a more professional comment here? Maybe we can research what these values are by looking how ABI encoding works in detail.

Reply:
```suggestion
        // Meta-data coming from event attributes of type bytes (cf. Ethereum RLP Encoding)
```